internal/metrics/querynode.go

```diff
@@ -0,0 +1,273 @@
+}
+
+func JoinQueryNodeID(label string, queryNodeID typeutil.UniqueID) string {
+	return typeutil.QueryNodeRole + strconv.FormatInt(queryNodeID, 10) + "_" + label
+}
+
+// queryNodeDurationBuckets involves durations in milliseconds
+var queryNodeDurationBuckets = []float64{0, 10, 100, 1000, 10000, 100000, 1000000, 10000000}
+
+var (
+	QueryNodeNumCollections = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "num_collections",
+			Help:      "Number of collections in QueryNode.",
+		}, []string{
+			"num_collections",
+		})
+
+	QueryNodeNumPartitions = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "num_partitions",
+			Help:      "Number of partitions in QueryNode.",
+		}, []string{
+			"num_partitions",
+		})
+
+	QueryNodeNumSegments = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "num_segments",
+			Help:      "Number of segments in QueryNode.",
+		}, []string{
+			"num_segments",
+		})
+
+	QueryNodeNumDmlChannels = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "num_dml_channels",
+			Help:      "Number of dmlChannels per collection in QueryNode.",
+		}, []string{
+			"num_dml_channels",
+		})
+
+	QueryNodeNumDeltaChannels = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "num_delta_channels",
+			Help:      "Number of deltaChannels per collection in QueryNode.",
+		}, []string{
+			"num_delta_channels",
+		})
+
+	QueryNodeNumConsumers = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "num_consumers",
+			Help:      "Number of consumers per collection in QueryNode.",
+		}, []string{
+			"num_consumers",
+		})
+
+	QueryNodeNumReaders = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "num_readers",
+			Help:      "Number of readers per collection in QueryNode.",
+		}, []string{
+			"num_readers",
+		})
+
+	QueryNodeSearchCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "search_count",
+			Help:      "Search_count request statistic in QueryNode.",
+		}, []string{
+			"status",
+		})
+
+	QueryNodeRetrieveCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "retrieve_count",
+			Help:      "Retrieve request statistic in QueryNode.",
+		}, []string{
+			"status",
+		})
+
+	QueryNodeQueryDuration = prometheus.NewHistogramVec(
```

Comment:
rename to QueryNodeSQReqLantency
and use label to distinguish search and query

Reply:
fixed